src/vm/eetwain.cpp

```diff
@@ -3805,6 +3806,28 @@ bool UnwindEbpDoubleAlignFrame(
     {
         TADDR baseSP;
 
+#ifdef WIN64EXCEPTIONS
+        // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+        // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+        // If funclet frame layout is changed from CodeGen::genFuncletProlog() and genFuncletEpilog(),
+        // we need to change here accordingly.
+        if (pCodeInfo->IsFunclet())
+        {
+            baseSP = curESP + 8; // padding for 16byte stack alignment allocated in genFuncletProlog()
```

Comment:
All the other funclet-based architectures also have fixed sized frames -- they pre-allocate the outgoing argument space for calls, and localloc/_alloca is not allowed in EH handlers / funclets by the CLI spec. So this is new territory.

One possibility is to set up ebp as a normal frame pointer in funclets, and then reserve and use some other register as the base pointer for accessing locals within funclet code generation. That would be a pretty pervasive change.

Another possibility would be to change to fixed frames. That doesn't seem great.

Or, as you suggest, you could reserve another (callee-saved) register for storing the funclet frame pointer. But that then becomes part of the ABI contract, as the VM would need to know which register that is.

There are so few registers already on x86 that reserving (another) one in funclets is going to hurt code quality.

I can't think of any obvious good solution right now.